src/app/services/resource.service.ts

```diff
@@ -259,4 +261,36 @@ export class ResourceService {
       });
   }
 
+  /**
+   * Gets all elements from the study dimension that satisfy the constaint if given
+   * @param {Constraint} constraint
+   * @returns {Observable<string[]>}
+   */
+  getStudyNames(constraint: Constraint): Observable<string[]> {
+    return this.transmartResourceService.getStudyNames(constraint)
+      .map((elements: TransmartStudyDimensionElements[]) => {
+        return TransmartMapper.mapTransmartStudyDimensionElements(elements);
+      });
+  }
+
+  /**
+   * Gets available dimensions for step 3
+   * @param studyNames
+   * @returns {Observable<string[]>}
+   */
+  getAvailableDimensions(studyNames: string[]): Observable<string[]> {
+    return this.transmartResourceService.getAvailableDimensions(studyNames)
```

Comment:
study names can be obtained from the studies inside constraint service

Reply:
but this has be a list only of studies that are satisfying all the constraints from previous steps